### ISSUE TITLE
Cobbler 3.2.0 fails to import ESXi image with a type error

### ISSUE THREAD

**Issue by mpsOxygen** [cobbler/cobbler#2448]
### Describe the bug

 cobbler fails to import ESXi image with am exception: 
Exception occured: <class 'TypeError'>
Exception value: cannot use a string pattern on a bytes-like object

### To Reproduce

1. mount ESXi iso: mount -t iso9660 -o loop,ro VMware-VMvisor-Installer-7.0b-16324942.x86_64.iso /mnt/esxi70b
2. try to import ESXi: cobbler import --name=ESXI70b --breed=vmware --path=/mnt/esxi70b

### Expected behavior

 I expected the import to succed.

### Version

````paste below
[root@lab-provisioning-cobbler ~]# cobbler version
Cobbler 3.2.0
  source: ?, ?
  build time: Sun Oct 25 00:00:00 2020
````

### Additional context

```
[root@lab-provisioning-cobbler ~]# cobbler import --name=ESXI70b --breed=vmware --path=/mnt/esxi70b
task started: 2020-11-03_153100_import
task started (id=Media import, time=Tue Nov  3 15:31:00 2020)
running python triggers from /var/lib/cobbler/triggers/task/import/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/import/pre/*
shell triggers finished successfully
Found a candidate signature: breed=vmware, version=esxi67
running: /usr/bin/file /var/www/cobbler/distro_mirror/ESXI70b/vmware-esx-base-osl.txt
received on stdout: /var/www/cobbler/distro_mirror/ESXI70b/vmware-esx-base-osl.txt: UTF-8 Unicode text, with very long lines, with CRLF, CR line terminators

received on stderr: 
Found a candidate signature: breed=vmware, version=esxi70
running: /usr/bin/file /var/www/cobbler/distro_mirror/ESXI70b/vmware-esx-base-osl.txt
received on stdout: /var/www/cobbler/distro_mirror/ESXI70b/vmware-esx-base-osl.txt: UTF-8 Unicode text, with very long lines, with CRLF, CR line terminators

received on stderr: 
Found a matching signature: breed=vmware, version=esxi70
Adding distros from path /var/www/cobbler/distro_mirror/ESXI70b:
running: /usr/bin/file /var/www/cobbler/distro_mirror/ESXI70b/tools.t00
received on stdout: /var/www/cobbler/distro_mirror/ESXI70b/tools.t00: gzip compressed data, was "tools-light.tar", last modified: Tue Jun  2 05:40:38 2020, from Unix, original size modulo 2^32 206807040

received on stderr: 
Exception occured: <class 'TypeError'>
Exception value: cannot use a string pattern on a bytes-like object
Exception Info:
  File "/usr/lib/python3.8/site-packages/cobbler/remote.py", line 98, in run
    rc = self._run(self)

  File "/usr/lib/python3.8/site-packages/cobbler/remote.py", line 296, in runner
    self.remote.api.import_tree(

  File "/usr/lib/python3.8/site-packages/cobbler/api.py", line 1489, in import_tree
    import_module.run(path, mirror_name, network_root, autoinstall_file, arch, breed, os_version)

  File "/usr/lib/python3.8/site-packages/cobbler/modules/managers/import_signatures.py", line 191, in run
    import_walker(self.path, self.distro_adder, distros_added)

  File "/usr/lib/python3.8/site-packages/cobbler/modules/managers/import_signatures.py", line 74, in import_walker
    func(arg, top, names)

  File "/usr/lib/python3.8/site-packages/cobbler/modules/managers/import_signatures.py", line 315, in distro_adder
    adtls.append(self.add_entry(dirname, kernel, initrd))

  File "/usr/lib/python3.8/site-packages/cobbler/modules/managers/import_signatures.py", line 341, in add_entry
    archs = self.learn_arch_from_tree()

  File "/usr/lib/python3.8/site-packages/cobbler/modules/managers/import_signatures.py", line 435, in learn_arch_from_tree
    import_walker(self.path, self.arch_walker, result)

  File "/usr/lib/python3.8/site-packages/cobbler/modules/managers/import_signatures.py", line 74, in import_walker
    func(arg, top, names)

  File "/usr/lib/python3.8/site-packages/cobbler/modules/managers/import_signatures.py", line 468, in arch_walker
    m = re_krn2.match(line)

!!! TASK FAILED !!!
```

**Comment by mpsOxygen** [comment]
I think the problem is in get_file_lines() from import_signatures.py, when it opens a gzip file it returns a bunch of bytes that will never work with  m = re_krn2.match(line)

**Comment by mpsOxygen** [comment]
Ok so the facts that get_file_lines() from import_signatures.py returns bytes is a bug, but I am also starting to think that using "kernel_arch":"tools\\.t00" in the signatures file for VMware is just wrong. That is an arhive with the vmware tools, so all executables not much text and there is nothing in there that could help you determine the architecture of the image. 

I will try and find a better option for determining the arch of an ESXi image and make a PR on the signatures file.

**Comment by SchoolGuy** [comment]
That information spares me a lot of time. I think that I can maybe reserve some time this weekend for this... But I am not sure.

**Comment by mpsOxygen** [comment]
I think vmware-esx-base-osl.txt would e a better fit for finding the arch. There are some included file names in there that contain architecture.

**Comment by SchoolGuy** [comment]
Okay so the weekend went differently then I anticipated. Looking at this when more time is available but we need to fix this... *sigh*

**Comment by SchoolGuy** [comment]
This is closed in favor of #2337
